File: backend/app/db/models/release_event.py
from datetime import datetime
from typing import Optional

from sqlalchemy import String, Text, ForeignKey, Boolean, DateTime
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base


class ReleaseEventType(Base):
    __tablename__ = "release_event_type"

    tenant_id: Mapped[int] = mapped_column(ForeignKey("tenant.id"), nullable=False, index=True)
    name: Mapped[str] = mapped_column(String(100), nullable=False)
    display_color: Mapped[Optional[str]] = mapped_column(String(7), nullable=True)
    is_system: Mapped[bool] = mapped_column(Boolean, nullable=False, default=False)
    deleted_at: Mapped[Optional[datetime]] = mapped_column(DateTime(timezone=True), nullable=True)


class ReleaseEvent(Base):
    __tablename__ = "release_event"

    tenant_id: Mapped[int] = mapped_column(ForeignKey("tenant.id"), nullable=False, index=True)
    release_id: Mapped[int] = mapped_column(
        ForeignKey("release.id", ondelete="CASCADE"), nullable=False, index=True
    )
    event_type_id: Mapped[int] = mapped_column(
        ForeignKey("release_event_type.id"), nullable=False, index=True
    )
    description: Mapped[str] = mapped_column(Text, nullable=False)
    occurred_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), nullable=False)
    recorded_by: Mapped[int] = mapped_column(ForeignKey("user.id"), nullable=False)

File: frontend/src/types/releaseEvent.ts
export interface ReleaseEventTypeResponse {
  id: number;
  tenant_id: number;
  name: string;
  display_color: string | null;
  is_system: boolean;
}

export interface ReleaseEventTypeCreatePayload {
  name: string;
  display_color?: string | null;
}

export interface ReleaseEventTypeUpdatePayload {
  name?: string;
  display_color?: string | null;
}

export interface ReleaseEventResponse {
  id: number;
  tenant_id: number;
  release_id: number;
  event_type_id: number;
  description: string;
  occurred_at: string;
  recorded_by: number;
}

export interface ReleaseEventCreatePayload {
  event_type_id: number;
  description: string;
  occurred_at?: string | null;
}
